### DISCUSSION TITLE
Strategy for ensuring URL always contains the version

### DISCUSSION THREAD

**Discussion by troypoulter** [facebook/docusaurus#8013] [Q&A] [answered]
## What I'm trying to do

I am looking to use Docusaurus for process documentation that goes through regular version updates: v2.0, v2.1, v2.2, v3.0 etc.

The links to the documentation will be stored in a form of attestation store, so that if I wanted to check back six months ago and see the reference material for that attestation, I want it to link to that exact version.

Ideally, I am after a way I can create a URL for the current version, like this: https://docusaurus.io/docs/2.0.1/versioning.

## An example

This is for a work project, so I don't have a reproducible example to share unfortunately, though I am very happy to make one if the below needs more explanation.

So we have this link in the official Docusaurus docs: https://docusaurus.io/docs/versioning and it's currently on v2.0.1.

![image](https://user-images.githubusercontent.com/19419349/187011970-79ccb6d0-b5aa-456f-b5be-4bc67f482886.png)

But if I was to access this link 6 months in the future, it would default to the current version. And as right now this is the current version, I can't produce a versioned URL, only for older ones e.g. https://docusaurus.io/docs/2.0.0-rc.1/versioning.

## Ideas on an approach

I was thinking to achieve this I could make the current version just a page saying "Please select a version via the drop-down" and then work out of the versioned docs folder. But thought to ask in case there are other strategies for this scenario.

**Comment by slorber** [comment, marked as answer]
You can configure the url path segment of each version independently, including the current version which defaults to path=`/`

https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-content-docs#VersionsConfig

```js
        docs: {
          versions: {
            current: {
              path "v2.0.1": 
            },
          },
        },
```

See also our versioning strategies guide: https://docusaurus.io/docs/versioning#configuring-versioning-behavior

---

Might be related: there's also this idea of "version aliases" (https://github.com/facebook/docusaurus/issues/3285) that we never implemented

**Reply by troypoulter** [reply]
Ahh, thanks for the help @slorber! That seemed to do the trick, I had to modify it slightly so that the version I want to default is there:

```json
          versions: {
            '2.0.1': {
              path: '2.0.1'
            },
          },
```

This will be great as it means each of the docs will have a version included in the URL.
